lib/streamlit/report_session.py

```diff
@@ -624,3 +608,52 @@ def _get_storage(self):
             else:
                 raise RuntimeError("Unsupported sharing mode '%s'" % sharing_mode)
         return self._storage
+
+
+def _populate_config_msg(msg):
+    msg.sharing_enabled = config.get_option("global.sharingMode") != "off"
+    msg.gather_usage_stats = config.get_option("browser.gatherUsageStats")
+    msg.max_cached_message_age = config.get_option("global.maxCachedMessageAge")
+    msg.mapbox_token = config.get_option("mapbox.token")
+    msg.allow_run_on_save = config.get_option("server.allowRunOnSave")
+
+    custom_theme_msg = msg.custom_theme
+
+    # TODO: Figure out what the exact behavior should be here after getting
+    # input from product. We'll probably want to specify a list of fields
+    # as required and warn or raise an error if required fields are only
+    # partially defined.
+    if config.get_option("customTheme.name"):
+        fields = [
+            "name",
+            "primary",
+            "secondary",
+            "sidebar",
+            "main",
+            "body_text",
+        ]
+        for field_name in fields:
+            field = config.get_option(f"customTheme.{to_lower_camel_case(field_name)}")
```

Comment:
Totally not a requirement at all but it would be cool if we had a get_options by section mostly for laziness if we do add more theme options. It's probably better this way so we can ensure there is intent to handle this in the same manner.

Reply:
Ah, yeah that would clean this up quite a bit / make it easier to add more options down the line. I added `config.get_options_by_section` (`get_option` and `get_options` seemed a bit too close) and reworked this code to use it.